sdk/core/azure-core/src/datetime.cpp

```diff
@@ -346,7 +346,7 @@ T ParseNumber(
 
   // ThrowParseError() will always throw, but there's no way to tell that to compiler, so a return
   // statement is required. It is not possible to cover the return line with tests.
-  return T(); // LCOV_EXCL_LINE
```

Comment:
Same here, I think such intentional exclusions where it is impossible to test should be allowed.

Reply:
Digging in a bit, the original comment is incorrect. The root issue was that the `ThrowParseError` was not annotated with the `[[noreturn]]` attribute which tells the compiler that the function will never return.

That renders the exclusion moot.

And in general, I agree with Rick: We either allow many exclusions or we allow none of them.

And allowing many exclusions means we have a false sense of what our *actual* code coverage numbers are.

Reply:
And I verified - the `return T()` line was totally unnecessary.

What I found while testing stuff and looking at coverage is that sometimes you have to get really creative.

But at the end of the day, I feel that it's more important that we have a code coverage number that we can trust than it is that we have a really high code coverage number.